src/executor/stack/executor.rs

```diff
@@ -1442,28 +1460,40 @@ impl<'config, 'precompiles, S: StackState<'config>, P: PrecompileSet> Handler
 	/// Provide a default implementation by fetching the code.
 	///
 	/// According to EIP-7702, the code size of an address is the size of the
-	/// delegated address code size.
+	/// `0xef01` - `2`.
+	/// <https://eips.ethereum.org/EIPS/eip-7702#delegation-designation>
 	fn code_size(&mut self, address: H160) -> U256 {
-		let target_code = self.authority_code(address);
-		U256::from(target_code.len())
+		//  If deleted account, return size equal to 2 bytes
+		if self.get_authority_target(address).is_some() {
+			U256::from(2)
+		} else {
+			let target_code = self.authority_code(address);
+			U256::from(target_code.len())
+		}
 	}
 
 	/// Fetch the code hash of an address.
 	/// Provide a default implementation by fetching the code.
 	///
 	/// According to EIP-7702, the code hash of an address is the hash of the
-	/// delegated address code hash.
+	/// `keccak256(0xef01): 0xeadcdba66a79ab5dce91622d1d75c8cff5cff0b96944c3bf1072cd08ce018329`.
+	/// <https://eips.ethereum.org/EIPS/eip-7702#delegation-designation>
 	fn code_hash(&mut self, address: H160) -> H256 {
+		// For EIP-7702 Hash of `EF01` bytes that is used for `EXTCODEHASH` when called from delegated address.
+		pub const EIP7702_MAGIC_HASH: [u8; 32] = [
```

Comment:
It might be nice to add a unit test checking that this constant is indeed equal to `keccak256(0xef01)`